subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/CurrentJvmToolchainSpec.java

```diff
@@ -20,8 +20,11 @@
 import org.gradle.internal.jvm.Jvm;
 import org.gradle.jvm.toolchain.JavaLanguageVersion;
 
+import javax.inject.Inject;
+
 public class CurrentJvmToolchainSpec extends DefaultToolchainSpec {
 
+    @Inject
```

Comment:
There are only a few places where the `CurrentJvmToolchainSpec` is called directly via constructor. After this PR is merged, we actually want to completely avoid usage of this spec directly and convert those usages to the calls to the toolchain service instead: `toolchainService.compilerFor {}` to benefit from the automatic fallback toolchain resolution for unconfigured toolchain specs.

As such, we should delay updating those usages until a follow-up PR, since some of the classes are not currently touched by this PR.